src/controllers/components/cart-panel.js

```diff
@@ -0,0 +1,37 @@
+import { searchItem, checkIn} from '../../lib/api-client';
+import { Dispatcher } from 'consus-core/flux';
+import StudentStore from '../../store/student-store';
+
+export default class CartController {
+
+    static checkInItem(id, itemAddress) {
+        return checkIn(id, itemAddress).then(item => {
+            Dispatcher.handleAction('CHECKIN_SUCCESS', {
+                itemAddress: item.itemAddress
+            });
+        }).catch(error => {
+            Dispatcher.handleAction('ERROR', {
+                error
+            });
+        });
+    }
+
+    static getItem(address) {
+        if(StudentStore.getStudent().hasOverdueItem)
+            return Dispatcher.handleAction('ERROR', {
+                error:'Student has at least one overdue item.'
+            });
+
+        return searchItem(address).then(item => {
+            if (item.status === 'CHECKED_OUT')
+                return Dispatcher.handleAction('ERROR', {
```

Comment:
Nit: I feel like this should be a WARN, but they are the exact same right now so it doesn't really matter.